lib/manageiq/active_record_connector.rb

```diff
@@ -0,0 +1,60 @@
+require "active_record"
+unless defined? DatabaseConfigurationPatch
+  require_relative "../patches/database_configuration_patch.rb"
+end
+
+module ManageIQ
+  class ActiveRecordConnector
+    # Helper class for getting the database config without needing Rails
+    #
+    # Makes use of the database_configuration_patch for the most part for
+    # grabbing this from the yaml file in config/database.yml
+    class ConnectionConfig
+      # Set a default method for handling super call in
+      # lib/patches/database_configuration_patch.rb
+      def self.database_configuration
+        if ENV["DATABASE_URL"]
+          {}
+        end
+      end
+
+      class << self
+        prepend DatabaseConfigurationPatch
+      end
+
+      def self.[](env)
+        database_configuration[env]
+      end
+    end
+
+    # Returns true if an ActiveRecord::Base.connection_config is present,
+    # otherwise returns false
+    def self.connection_exists?
+      begin
+        true if ActiveRecord::Base.connection_config
```

Comment:
@jrafanie I may not be able to use that method.  Looks like when you call `.connected?` prior to having a connection, it creates a new `connection_handler` object when it does, and then when `.establish_connection` is then called and `.connected?` is called again, it is still `false`.

I will have to do some more testing, but it seems like the `connection_handler` registry isn't being updated when calling `establish_connection`, but I could be wrong.